price-server/src/app/mod.rs

```diff
@@ -1,43 +1,137 @@
 mod error;
 
+use std::{collections::HashMap, sync::Arc};
+
 use chrono::Duration;
 use futures::stream::StreamExt;
+use rust_decimal::Decimal;
+use tokio::sync::RwLock;
 use tracing::{info_span, instrument, Instrument};
 
-use shared::{health::HealthCheckTrigger, payload::OkexBtcUsdSwapPricePayload, pubsub::*};
-
-use super::exchange_price_cache::ExchangePriceCache;
+use shared::{
+    exchanges_config::ExchangeConfigAll,
+    health::HealthCheckTrigger,
+    payload::{
+        KolliderBtcUsdSwapPricePayload, OkexBtcUsdSwapPricePayload, KOLLIDER_EXCHANGE_ID,
+        OKEX_EXCHANGE_ID,
+    },
+    pubsub::*,
+};
 
 pub use crate::{currency::*, fee_calculator::*};
+use crate::{
+    exchange_tick_cache::ExchangeTickCache,
+    price_mixer::{PriceMixer, PriceProvider},
+};
 pub use error::*;
 
 pub struct PriceApp {
-    price_cache: ExchangePriceCache,
+    price_mixer: PriceMixer,
     fee_calculator: FeeCalculator,
 }
 
 impl PriceApp {
     pub async fn run(
-        mut health_check_trigger: HealthCheckTrigger,
+        health_check_trigger: HealthCheckTrigger,
         fee_calc_cfg: FeeCalculatorConfig,
         pubsub_cfg: PubSubConfig,
+        exchanges_cfg: ExchangeConfigAll,
     ) -> Result<Self, PriceAppError> {
+        let health_check_trigger = Arc::new(RwLock::new(health_check_trigger));
+
+        let mut providers: HashMap<String, (Box<dyn PriceProvider + Sync + Send>, Decimal)> =
+            HashMap::new();
+
+        if let Some(config) = exchanges_cfg.kollider.as_ref() {
+            let kollider_price_cache = ExchangeTickCache::new(Duration::seconds(30));
```

Comment:
should exctract the `cache` time into a constant or a config setting. Using the same magic number here and line 60 is not good.